### ISSUE TITLE
How is it different than [[piroor/treestyletab: Tree Style Tab, Show tabs like a tree.](https://github.com/piroor/treestyletab)](https://github.com/piroor/treestyletab)?

### ISSUE THREAD

**Issue by anishmittal2020** [mbnuqw/sidebery#587]
How is it different than [[piroor/treestyletab: Tree Style Tab, Show tabs like a tree.](https://github.com/piroor/treestyletab)](https://github.com/piroor/treestyletab)?

**Comment by OJFord** [comment]
Primarily panels, which afaik TST doesn't have? (And associated features, like assigning them to a FF container.)

**Comment by Mckol** [comment]
- **Snapshots** are a major feature. If you have a lot of tabs open and always restore browser sessions, losing the tree structure can be really painful. With Sidebery, you can just restore the last snapshot if that happens and basically never need to worry about it. With TST, IIRC I had to use a separate extension for this which required extra configuration to save the tree structure.
- Also, in my opinion Sidebery **looks** much better out-of-the-box.
- Tree Style Tab seems to rely on extensions for a bunch of features (look at their AMO page which lists some of them) while Sidebery just includes everything in itself. Whether this is good or bad depends on what you want.

**Comment by carlgieringer** [comment]
Can someone please explain the intended use-case(s) for Panels?

Here is my situation: I am a TreeStyleTab user with five windows open and around 100 tabs per window . Every time I wake my computer, I experience a slowdown as memory pressure goes red in Activity Monitor, and I see that Firefox is using 4GB RAM, FirefoxCP WebExtensions 6GB RAM, and FirefoxCP Isolated Web Content using ~6GB across all processes. (Computer has 16GB RAM.) I need to decrease my tabs, but I can't stand to lose my precious browsing state.

So I head over to piroor/treestyletab to look for a session manager. I tend to treat my windows like projects and my trees of tabs like trains-of-thought within that project. I have pinned tabs that represent the landing page(s) or primary editors/artifacts of the project, too. What I need to be able to do is to put a project away for awhile and unload its tabs. But I want to be able to pick up on a project at a later point.

My requirements:

* Will store all the tabs from a window (or Panel) at once and restore them later. (Required)
* Allows me to name the stored sessions (Strong ask.)
* Allows me to add tabs from active windows to stored sessions and/or to move tabs between stored session windows (nice to have.)

So I discover Sidebery and figure I'll give it a go. Like other commenters, I have a good initial experience. And I discover this Panel feature that I don't quite understand. Here's what I want to use them for (is this a mismatch with the intended functionality?):

I want to combine all my separate windows into one window with the tabs from each window in its own Panel. (I am _really_ tired of cycling through windows trying to find the one for a particular project. This gets extra confusing when I sometimes have similar tabs open in multiple windows.) I will name the Panel after the project. For session storage, I want to store/restore the panels as a group (is this possible? I see a "Zzz unload tabs" button in the context menu for Panels.)

One pain point I encounter immediately is that it is not possible to send all the tabs from the default Panel in one window to a new or existing Panel (other than 'default') in another window. My workaround is that I don't allow any tabs in the 'default' Panel of my 'chosen' one window, and instead use it like a staging ground. First I move all the tabs from a window to this window's empty 'default' Panel, and then I move them all from there into a new Panel in that window.

I also see some features of Panels that suggest to me that I am misusing them as unloadable 'projects':

1. All Panels are available in all windows. I never see myself needing this feature, and it actually goes against my workflow because I want all my tabs related to one project in the same place. If tabs from a project are spread across multiple windows, that would be really confusing.

    This feature suggests to me that Panels are less like projects and more like modes that a person can be in regardless of what they are doing. Maybe an example of that might be like "put all stackoverflow tabs into this Panel because that is 'work'" (I wouldn't like to do that, but it seems to relate to the other feature)
4. Panels have features like "Move tab to this panel if it is opened in this container" and "Move tabs with matched URL to this panel". This seems like Panels are for the same things as Firefox Containers, which based upon the default containers would be things like "work", "shopping", and "personal".

So...what are the recommended ways to use Panels? I'm nervous to build up a work flow on them that goes against the developer's plans, because they may change or remove features on which I depend if I am using them differently than intended.

Thanks!

**Comment by OJFord** [comment]
@carlgieringer 

1. I think project per panel is fine, yes it makes it easier to switch 'modes of working'/project - but you don't have to? Do you even need multiple windows still if you use Sidebery? Only if you need two tabs open side by side I suppose. For me that would be a brief temporary thing, the second window would never have a second tab.
1. Panels don't have to be assigned containers.

**Comment by mYnDstrEAm** [comment]
I loved TST (btw a nice thing about it is how extendable it is with other addons like the TST bookmarks panel) but now it seems to be causing Firefox freezes (https://github.com/piroor/treestyletab/issues/3816) slowing down my browsing a lot (more than a lot) which is why I just switched to sideberry. A nice thing is that the tree structure built with TST is kept so one can easily migrate back and forth - it may be a good idea to mention this in the readme. If it was I would have tried this addon much sooner as I thought I would need to do something to keep my tree structure if it's possible at all.

One thing I don't like at all about sideberry is how the trees are uncollapsed by default and it took me a while to find out how to collapse the branch. It may be better to display the un/collapse icon before the favicon for branch tabs or to display a hint after installation once the first branch is displayed so the user learns this early.
Long story short: performance / RAM issues could be one of the largest differences. I've also heard other users say sideberry needs less RAM and works more fluently (not an advantage if you don't have such problems with TST).
